paimon-spark/paimon-spark-common/src/test/java/org/apache/paimon/spark/SparkSchemaEvolutionITCase.java

```diff
@@ -276,7 +276,72 @@ public void testDropPrimaryKey() {
                 .satisfies(
                         anyCauseMatches(
                                 UnsupportedOperationException.class,
-                                "Cannot drop/rename primary key[b]"));
+                                "Cannot drop partition key or primary key: [b]"));
+    }
+
+    @Test
+    public void testRenamePrimaryKey() {
+        spark.sql(
+                "CREATE TABLE test_rename_primary_key_table (\n"
+                        + "a BIGINT NOT NULL,\n"
+                        + "b STRING)\n"
+                        + "TBLPROPERTIES ('primary-key' = 'a')");
+
+        spark.sql("INSERT INTO test_rename_primary_key_table VALUES(1, 'aaa'), (2, 'bbb')");
+
+        spark.sql("ALTER TABLE test_rename_primary_key_table RENAME COLUMN a to a_");
+
+        List<Row> result =
+                spark.sql("SHOW CREATE TABLE test_rename_primary_key_table").collectAsList();
+        assertThat(result.toString())
+                .contains(
+                        showCreateString(
+                                "test_rename_primary_key_table", "a_ BIGINT NOT NULL", "b STRING"))
+                .contains("'primary-key' = 'a_'");
+
+        List<String> actual =
+                spark.sql("SELECT * FROM test_rename_primary_key_table").collectAsList().stream()
+                        .map(Row::toString)
+                        .collect(Collectors.toList());
+
+        assertThat(actual).containsExactlyInAnyOrder("[1,aaa]", "[2,bbb]");
+
+        spark.sql("INSERT INTO test_rename_primary_key_table VALUES(1, 'AAA'), (2, 'BBB')");
```

Comment:
This refers to the actual column projection used when reading data files during the debugging process.

This refers to data written before rename, the projection columns are `_KEY_a` and `a`:
![Screenshot 2024-08-04 at 21 51 16](https://github.com/user-attachments/assets/acfaaeff-667f-463e-9fa2-9818394a6fa9)

This refers to data written after rename, the projection columns are `_KEY_a_` and `a_`:
![Screenshot 2024-08-04 at 22 11 52](https://github.com/user-attachments/assets/8df1dace-703a-43ee-bc3a-838c191859e5)